src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -0,0 +1,320 @@
+                    OperationKind.PropertyReference,
+                    OperationKind.FieldReference,
+                    OperationKind.DelegateCreation,
+                    OperationKind.EventReference,
+                    OperationKind.Unary,
+                    OperationKind.Binary,
+                    OperationKind.ArrayCreation,
+                    OperationKind.CatchClause
+                    );
+
+                // Handle library side definitions of preview features
+                context.RegisterSymbolAction(context => AnalyzeSymbol(context, requiresPreviewFeaturesSymbols, requiresPreviewFeaturesSymbolsToUsageType, runtimeFeatureType), s_symbols);
+            });
+        }
+
+        private static bool TypeSymbolHasAttribute(ITypeSymbol symbol, ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols, ConcurrentDictionary<ISymbol, PreviewFeatureUsageType> requiresPreviewFeaturesSymbolsToUsageType)
+        {
+            ImmutableArray<INamedTypeSymbol> interfaces = symbol.Interfaces;
+            foreach (INamedTypeSymbol? anInterface in interfaces)
+            {
+                var interfaceMembers = anInterface.GetMembers();
+                if (interfaceMembers.Length == 0)
+                {
+                    // Only tag empty interfaces to prevent breaking changes in the future
+                    requiresPreviewFeaturesSymbolsToUsageType.GetOrAdd(symbol, PreviewFeatureUsageType.EmptyInterface);
+                    return true;
+                }
+            }
+
+            INamedTypeSymbol? baseType = symbol.BaseType;
+            if (baseType != null)
+            {
+                return HasPreviewAttribute(baseType, requiresPreviewFeaturesSymbols);
+            }
+
+            return false;
+        }
+
+        private static bool ContainingTypeIsPreview(ISymbol symbol, ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols)
+        {
+            INamedTypeSymbol? containingType = symbol.ContainingType;
+            if (containingType != null)
+            {
+                return HasPreviewAttribute(containingType, requiresPreviewFeaturesSymbols);
+            }
+
+            return false;
+        }
+
+        private static bool SymbolIsStaticAndAbstract(ISymbol symbol, ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols)
+        {
+            bool isStaticAndAbstract = symbol.IsStatic && symbol.IsAbstract;
+
+            return isStaticAndAbstract && !ContainingTypeIsPreview(symbol, requiresPreviewFeaturesSymbols);
+        }
+
+        private static bool PropertyOrMethodSymbolHasAttribute(ISymbol propertyOrMethodSymbol, ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols, ConcurrentDictionary<ISymbol, PreviewFeatureUsageType> requiresPreviewFeaturesSymbolsToUsageType, INamedTypeSymbol? runtimeFeatureType)
+        {
+            if (SymbolIsStaticAndAbstract(propertyOrMethodSymbol, requiresPreviewFeaturesSymbols))
```

Comment:
Unfortunately we can't test this yet. `public abstract static void Foo()` will not compile with the current version of the compiler. Once the new compiler is consumable in an sdk build, I will update the branch with that build and add new unit tests

Reply:
Did you try to test by setting the `LanguageVersion`? https://github.com/dotnet/roslyn-analyzers/blob/a0522afd52b25ad211298e308fd5bbc7734865e0/src/Roslyn.Diagnostics.Analyzers/UnitTests/DoNotCopyValueTests.cs#L25
Not sure if the version you need available there, though it has even preview option https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/LanguageVersion.cs,198

Reply:
> Did you try to test by setting the LanguageVersion

I don't think that will work yet. To use `abstract static` we will need to start consuming the latest compiler from the latest sdk itself. Once this PR goes in, I will update this branch to consume the latest daily build of the sdk and then I can add a new unit test for this code path.